src/utils/logger/file.rs

```diff
@@ -59,16 +59,28 @@ impl InternalFileLogger {
     /// Rotate the log files.
     fn rotate(&mut self) -> io::Result<()> {
         for i in 0..self.rotations - 1 {
-            let from = format!("{}.{}", &self.path, self.rotations - i - 1);
-            let to = format!("{}.{}", &self.path, self.rotations - i);
+            let mut from = self.path.as_os_str().to_os_string();
+            let mut to = self.path.as_os_str().to_os_string();
 
-            if Path::new(&from).exists() {
+            from.push(format!(".{}", self.rotations - i - 1));
+            to.push(format!(".{}", self.rotations - i));
+
+            let from = PathBuf::from(from);
+            let to = PathBuf::from(to);
+
+            if from.exists() {
                 fs::rename(&from, &to)?;
             }
         }
 
         if self.rotations > 0 {
-            fs::rename(&self.path, &format!("{}.1", &self.path))?;
+            let mut to = self.path.as_os_str().to_os_string();
+
+            to.push(".1");
+
+            let to = PathBuf::from(to);
```

Comment:
Why not
```rust
let mut to = self.path.clone();
to.push(".1");
```

Reply:
`PathBuf::push()` does a different thing. It extends the current path with another path segment, so if `self.path` was for example `/var/log/arrow-client.log`, then after calling the `to.push(".1")`, we'd get `/var/log/arrow-client.log/.1` instead of the desired `/var/log/arrow-client.log.1`.

We also cannot use `with_extension`/`set_extension` here because it'd replace any possibly existing extension with a given one.

Reply:
Ahh. Thanks for clarification.